### PR TITLE
[NSSecureCoding] Enable nullability and a few other code improvements.

### PR DESCRIPTION
**Pull request by rolfbjarne**

### CODE REVIEW

**Review by dalexsoto — APPROVED** [review]

**Review by mandel-macaque — APPROVED** [review]

File: src/Foundation/NSSecureCoding.cs [hunk 1]
```diff
@@ -19,8 +21,8 @@ public partial class NSSecureCoding {
 
 		public static bool SupportsSecureCoding (Type type)
 		{
-			if (type == null)
-				throw new ArgumentNullException ("type");
+			if (type is null)
+				throw new ArgumentNullException (nameof (type));
```
Comment (spouliot) [inline src/Foundation/NSSecureCoding.cs:25]: Better use `ObjCRuntime.ThrowHelper.ThrowArgumentNullException ((nameof (type));`
Throwing an exception generally means the method can't be inlined.

File: src/Foundation/NSSecureCoding.cs [hunk 2]
```diff
@@ -39,8 +41,8 @@ public static bool SupportsSecureCoding (Type type)
 
 		public static bool SupportsSecureCoding (Class klass)
 		{
-			if (klass == null)
-				throw new ArgumentNullException ("klass");
+			if (klass is null)
+				throw new ArgumentNullException (nameof (klass));
```
Comment (spouliot) [inline src/Foundation/NSSecureCoding.cs:45]: same

**Review by spouliot — CHANGES_REQUESTED** [review]

File: src/Foundation/NSSecureCoding.cs (hunk 1 above)
Comment (rolfbjarne) [reply src/Foundation/NSSecureCoding.cs]: Fixed.

File: src/Foundation/NSSecureCoding.cs (hunk 2 above)
Comment (rolfbjarne) [reply src/Foundation/NSSecureCoding.cs]: Fixed.

**Review by tj-devel709 — APPROVED** [review]

**Comment by rolfbjarne** [comment]
Test failures are unrelated:

* monotouch-test/Mac [dotnet]/Debug [dotnet]: https://github.com/xamarin/maccore/issues/2537
* dont link/Mac Catalyst [dotnet]/Release [dotnet]: https://github.com/xamarin/maccore/issues/2414

**Pull request merged**